### PR TITLE
Add simple simulation tracing

### PR DESCRIPTION
**Pull request by mcches**
Here is an example trace ->

```txt
2022-10-20T21:20:02.325402Z  INFO turmoil: New server @127.0.0.1
2022-10-20T21:20:02.325633Z  INFO turmoil: New client @127.0.0.2
2022-10-20T21:20:02.325828Z  INFO turmoil: Bind 127.0.0.1:1738 UDP
2022-10-20T21:20:02.325937Z  INFO turmoil: Bind 127.0.0.2:1738 UDP
2022-10-20T21:20:02.325986Z  INFO turmoil: Send 127.0.0.2:1738 127.0.0.1:1738 UDP [0x70, 0x69, 0x6E, 0x67]
2022-10-20T21:20:02.328982Z  INFO turmoil: Delivered 127.0.0.1:1738 127.0.0.2:1738 UDP [0x70, 0x69, 0x6E, 0x67]
2022-10-20T21:20:02.329067Z  INFO turmoil: Recv 127.0.0.1:1738 127.0.0.2:1738 UDP [0x70, 0x69, 0x6E, 0x67]
2022-10-20T21:20:02.329092Z  INFO turmoil: Send 127.0.0.1:1738 127.0.0.2:1738 UDP [0x70, 0x6F, 0x6E, 0x67]
2022-10-20T21:20:02.329122Z  INFO turmoil: Unbind 127.0.0.1:1738 UDP
2022-10-20T21:20:02.329749Z  INFO turmoil: Delivered 127.0.0.2:1738 127.0.0.1:1738 UDP [0x70, 0x6F, 0x6E, 0x67]
2022-10-20T21:20:02.329809Z  INFO turmoil: Recv 127.0.0.2:1738 127.0.0.1:1738 UDP [0x70, 0x6F, 0x6E, 0x67]
2022-10-20T21:20:02.329832Z  INFO turmoil: Unbind 127.0.0.2:1738 UDP
```

### CODE REVIEW

**Comment by mcches** [comment]
We can run reverse dns lookup for every trace as well. I left it out as it requires a bit more wiring into top.

**Review by marcbowes — APPROVED** [review]

File: src/world.rs [hunk 1]
```diff
@@ -107,6 +109,8 @@ impl World {
     /// Send `message` from `src` to `dst`. Delivery is asynchronous and not
     /// guaranteed.
     pub(crate) fn send_message(&mut self, src: SocketAddr, dst: SocketAddr, message: Protocol) {
+        trace!("Send {} {} {}", src, dst, message);
```
Comment (davidbarsky) [inline src/world.rs:112]: small nit: i'd rephrase this as:

```rust
trace!(?src, ?dst, ?message, "sending message");
```

this allows those values to be emitted as structured key/value pairs.

File: src/world.rs [hunk 2]
```diff
@@ -95,6 +95,8 @@ impl World {
             "already registered host for the given socket address"
         );
 
+        trace!("New {} @{}", self.dns.reverse(addr), addr);
+
```
Comment (davidbarsky) [inline src/world.rs:99]: another nit:

```rust
trace!(src = ?self.dns.reverse(addr), dst = ?addr "new connection");
```

**Review by davidbarsky — COMMENTED** [review]
few small nits. arguably, these events aren't really events—they're probably better represented as spans.
